87.md

```diff
@@ -0,0 +1,120 @@
+
+```json
+{
+  "kind": 35834,
+  "content": "",
+  "tags": [
+    ["d", "<uuid>"],
+    ["name", "<Group name>"],
+    ["about", "<Group description>"],
+    ["picture", "<Group thumnail>"],
+    ["banner", "<Group banner image>"],
+    ["relay", "<url>"],
+    ["relay", "<url>"]
+  ]
+}
+```
+
+## Admin Key
+
+The key used to publish the group's `kind 35834` definition event MUST be used for performing privileged actions, such as publishing key rotations. Non-administrative events MAY be published to the group by the admin.
+
+Because the admin key is used for managing key rotation, it's recommended that a dedicated key be used for group administration instead of the group admin's personal key.
+
+## Key Sharing
+
+Key sharing and rotation is accomplished as described in [86.md](NIP 86), with the addition to the `kind 24` event of an `a` tag pointing to the group's address. Admins MAY share admin keys with themselves (for use on multiple devices) or with others the same way. In either case, the `role` tag MUST be used to specify whether the key is a `member` key or an `admin` key.
+
+```json
+{
+  "kind": 24,
+  "content": "",
+  "tags": [
+    ["a", "35834:<admin pubkey>:<group name>"],
+    ["privkey", "<the private key being shared>"],
+    ["relay", "<where to find group messages"],
+    ["role", "<member|admin>"]
+  ]
+}
+```
+
+## Access Requests
+
+Anyone may request access to a group using a `kind 25` rumor wrapped with an ephemeral key and addressed to the admin's key. The rumor's `content` MAY include a message, and its `tags` MUST include an `a` tag pointing to the group definition event's address. `tags` MAY also include a `claim` which admins can use to validate the access request, for example an invite code or payment receipt.
+
+On receipt, an admin MAY choose to admit the pubkey either by rotating keys or sharing the most recent shared key.
+
+```json
+{
+  "kind": 25,
+  "content": "Pleeease let me in",
+  "tags": [
+    ["a", "35834:<admin pubkey>:<group name>"],
+    ["claim", "NOSTR2023"]
+  ]
+}
+```
+
+## Exit Requests
+
+Anyone may request to be removed from a group using a `kind 26` rumor wrapped with an ephemeral key and addressed to the admin's key. The rumor's `content` MAY include a message, and its `tags` MUST include an `a` tag pointing to the group definition event's address.
+
+On receipt, an admin SHOULD remove the pubkey by rotating keys.
+
+```json
+{
+  "kind": 26,
+  "content": "I'm outta here",
+  "tags": [["a", "35834:<admin pubkey>:<group name>"]]
+}
+```
+
+## Messages
+
+Any group member MAY post an event to the group either publicly following NIP 72, or privately. Private events MUST be sealed by the user's own key, wrapped by the shared key, and addressed to the shared key. Anyone with the shared key may decrypt these messages.
```

Comment:
```suggestion
Any group member MAY post an event to the group either publicly following NIP 72, or privately. Private events MUST be sealed by the user's own key, wrapped by the shared key, and addressed to nobody. Anyone with the shared key may decrypt these messages.
```
If the wrapper's pubkey is the shared one, members can already filter by author instead of by #p.